### PR TITLE
breaking: update to OpenLLM 0.6

### PR DESCRIPTION
**Pull request by aarnphm**
# Description

Update the base classes to OpenLLM 0.6 architecture, which this, we deprecate `OpenLLMAPI` and make `OpenLLM` a thin wrapper around OpenAI.

`OpenLLM` class would now behave similar to OpenAI class.

Signed-off-by: Aaron Pham <contact@aarnphm.xyz>

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by logan-markewich** [comment]
Hmm, I'm not able to push to this PR for some reason

To fix the cicd, ctrl+f for `llama-index-integrations/llms/llama-index-llms-openllm:poetry#openllm-client` and remove those lines (there's like 8 of them)

**Comment by aarnphm** [comment]
Hi @logan-markewich, this should be fixed. I will also updated the permission on the fork.

File: llama-index-integrations/llms/llama-index-llms-openllm/llama_index/llms/openllm/base.py
[the remark this answers is not in the document]
Comment (aarnphm) [reply llama-index-integrations/llms/llama-index-llms-openllm/llama_index/llms/openllm/base.py:1]: Correct me if im wrong but isn't openai-like class a thin wrapper around the original `OpenAI` class?

[the remark this answers is not in the document]
Comment (logan-markewich) [reply llama-index-integrations/llms/llama-index-llms-openllm/llama_index/llms/openllm/base.py:1]: Thanks!

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**